src/server.rs

```diff
@@ -86,23 +109,21 @@ async fn handle_measurement_submit(
       parse_message(&bincode_msg)?;
 
       if let Some(min_revision) = state.min_revision_map.get(channel_name) {
-        let req_revision = request
-          .headers()
-          .get(HeaderName::from_static(REVISION_HEADER))
-          .map(|v| {
-            v.to_str()
-              .unwrap_or_default()
-              .parse::<usize>()
-              .unwrap_or_default()
-          })
-          .unwrap_or_default();
+        let req_revision: usize =
+          extract_and_parse_header(&request, REVISION_HEADER).unwrap_or_default();
         if req_revision < *min_revision {
           // Just ignore older requests gracefully
           return Ok(HttpResponse::NoContent().finish());
         }
       }
+      let threshold: Option<usize> = extract_and_parse_header(&request, THRESHOLD_HEADER);
+      if let Some(threshold) = threshold {
+        if !state.request_threshold_range.contains(&threshold) {
+          return Err(WebError::BadThreshold);
```

Comment:
Ok, the server only accepts threshold values within whatever range we set (between 20 and 50 by default) so there's a bound to how many queues it will build. Excellent.